mos-platform/atari2600-3e/mapper_3e.c

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 LLVM-MOS Project
+// Licensed under the Apache License, Version 2.0 with LLVM Exceptions.
+// See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+// information.
+
+#include <peekpoke.h>
+#include <mapper.h>
+
+__attribute__((section(".zp.bss"))) bank_index_t __current_bank;
+
+__attribute__((weak)) void set_current_bank(bank_index_t bank_id) {
+  if (bank_select && !(bank_id & 0x80)) {
+    bank_select(bank_id);
+  } else if (ram_select) {
+    ram_select(bank_id & 0x7f);
+  }
+}
+
+__attribute__((always_inline, leaf)) bank_index_t get_current_bank(void) {
```

Comment:
Leaf does nothing on a function definition; it's only use is on declarations of functions external to a LTO (link-time optimization) unit.

Also always_inline is generally not a good idea unless there's a compelling reason; the inliner generally does a good job of deciding whether to inline cases like these, and always_inline can interfere with e.g. taking the address of the function.

Reply:
Actually, I rescind the always_inline part of this; it looks like on clang at least it's not a failure if it can't actually inline; always_inline just makes it always try.